cdap-runtime-ext-dataproc/src/main/java/co/cask/cdap/runtime/spi/provisioner/dataproc/DataProcClient.java

```diff
@@ -246,11 +246,11 @@ public OperationSnapshot createCluster(String name, String imageVersion)
           requiredPorts.clear();
           addTag = true;
         } else if ("tcp".equals(protocol)) {
-          if (allowed.getPorts().contains(String.valueOf(FirewallPort.HTTPS.port))) {
+          if (allowed.getPorts() != null && allowed.getPorts().contains(String.valueOf(FirewallPort.HTTPS.port))) {
```

Comment:
if the ports are empty, does that mean all ports are open? If so, this should be `allowed.getPorts() == null || ....`

Reply:
I looked through some documentation as well as the GCP UI (VPC Network), but was unable to determine what empty ports indicates.
I do think that it would indicate that all ports are open, though, since otherwise specifying just "tcp" as a rule would not have meaning.